lib/child_process.js

```diff
@@ -245,39 +243,29 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    if (encoding)
-      child.stdout.setEncoding(encoding);
-
     child.stdout.addListener('data', function(chunk) {
-      stdoutLen += chunk.length;
+      stdoutLen += (chunk.byteLength || chunk.length);
```

Comment:
Ah, yes, that alternative does have the virtue of counting bytes correctly.

I'm inclined to not keep the original `setEncoding()` call in `child_process.js` because it means we're converting Buffer to string and then back to Buffer again rather than just doing Buffer to string. I imagine that would have performance implications... But, uh, I didn't benchmark that, so uh... ¯\\_(ツ)_/¯